### PR TITLE
Catch Piwik::ApiError when segment data hasn't been processed

### PR DESCRIPTION
**Pull request by felipecabargas**
Fixes [monsido_app/#3877](https://github.com/Monsido/monsido_app/issues/3877)

### CODE REVIEW

File: lib/piwik/base.rb [hunk 1]
```diff
@@ -152,17 +153,19 @@ def call(method, params, piwik_url=nil, auth_token=nil)
         $VERBOSE = verbose_obj_save
         if xml.is_a?(String) && xml.force_encoding('BINARY').is_binary_data?
           xml.force_encoding('BINARY')
+        elsif xml =~ /error message="These reports have no data, because the Segment you requested (direct) has not yet been processed by the system/
```
Comment (twarberg) [inline lib/piwik/base.rb]: I think `(direct)` is request specific

Comment (twarberg) [reply lib/piwik/base.rb]: eg. could be something else on another request

**Review by serch — APPROVED** [review]

**Comment by twarberg** [comment]
@felipecabargas It needs to be against `new_features` I've changed and theres a conflict

**Comment by felipecabargas** [comment]
@twarberg fixed !

**Comment by serch** [comment]
closing stale PR, will reopen if necessary

**Pull request closed without merging**